unittests/rpc/handlers/AMMInfoTests.cpp

```diff
@@ -0,0 +1,1154 @@
+
+    ON_CALL(*backend, fetchLedgerBySequence).WillByDefault(Return(lgrInfo));
+    ON_CALL(*backend, doFetchLedgerObject(accountKey, testing::_, testing::_))
+        .WillByDefault(Return(accountRoot.getSerializer().peekData()));
+    ON_CALL(*backend, doFetchLedgerObject(ammKeylet.key, testing::_, testing::_))
+        .WillByDefault(Return(std::optional<Blob>{}));
+
+    auto static const input = json::parse(fmt::format(
+        R"({{
+            "amm_account": "{}"
+        }})",
+        AMM_ACCOUNT
+    ));
+
+    auto const handler = AnyHandler{AMMInfoHandler{backend}};
+    runSpawn([&](auto yield) {
+        auto const output = handler.process(input, Context{yield});
+        ASSERT_FALSE(output);
+
+        auto const err = rpc::makeError(output.error());
+        EXPECT_EQ(err.at("error").as_string(), "actNotFound");
+        EXPECT_EQ(err.at("error_message").as_string(), "Account not found.");
+    });
+}
+
+TEST_F(RPCAMMInfoHandlerTest, AMMAccountAccBlobNotFound)
+{
+    backend->setRange(10, 30);
+
+    auto const lgrInfo = CreateLedgerInfo(LEDGERHASH, 30);
+    auto const accountKey = GetAccountKey(AMM_ACCOUNT);
+    auto const account2Key = GetAccountKey(AMM_ACCOUNT2);
+    auto const ammId = ripple::uint256{AMMID};
+    auto const ammKeylet = ripple::keylet::amm(ammId);
+
+    auto accountRoot = CreateAccountRootObject(AMM_ACCOUNT, 0, 2, 200, 2, INDEX1, 2);
+    auto const ammObj =
+        CreateAMMObject(AMM_ACCOUNT2, "XRP", ripple::toBase58(ripple::xrpAccount()), "JPY", AMM_ACCOUNT2);
+    accountRoot.setFieldH256(ripple::sfAMMID, ammId);
+
+    ON_CALL(*backend, fetchLedgerBySequence).WillByDefault(Return(lgrInfo));
+    ON_CALL(*backend, doFetchLedgerObject(accountKey, testing::_, testing::_))
+        .WillByDefault(Return(accountRoot.getSerializer().peekData()));
+    ON_CALL(*backend, doFetchLedgerObject(ammKeylet.key, testing::_, testing::_))
+        .WillByDefault(Return(ammObj.getSerializer().peekData()));
+    ON_CALL(*backend, doFetchLedgerObject(account2Key, testing::_, testing::_))
+        .WillByDefault(Return(std::optional<Blob>{}));
+
+    auto static const input = json::parse(fmt::format(
+        R"({{
+            "amm_account": "{}"
+        }})",
+        AMM_ACCOUNT
+    ));
+
+    auto const handler = AnyHandler{AMMInfoHandler{backend}};
+    runSpawn([&](auto yield) {
+        auto const output = handler.process(input, Context{yield});
+        ASSERT_FALSE(output);
+
+        auto const err = rpc::makeError(output.error());
+        EXPECT_EQ(err.at("error").as_string(), "actNotFound");
+        EXPECT_EQ(err.at("error_message").as_string(), "Account not found.");
+    });
+}
+
+TEST_F(RPCAMMInfoHandlerTest, HappyPathMinimalFirstXRPNoTrustline)
+{
+    backend->setRange(10, 30);
+
+    auto const account1 = GetAccountIDWithString(AMM_ACCOUNT);
+    auto const account2 = GetAccountIDWithString(AMM_ACCOUNT2);
+    auto const lgrInfo = CreateLedgerInfo(LEDGERHASH, SEQ);
```

Comment:
To make test a bit shorter you could move all the duplicated data into some new fixture inherited from `RPCAMMInfoHandlerTest`.

Reply:
Yes but then you can't see exactly what each test uses and does. Sure there is a lot of duplication here but sometimes that might be better than hiding stuff in base-classes. WDYT?